sunpy/coordinates/ephemeris.py

```diff
@@ -0,0 +1,230 @@
+
+    Returns
+    -------
+    out : `~astropy.coordinates.Longitude`
+        The Carrington longitude
+    """
+    obstime = _astropy_time(time)
+
+    # Calculate the longitude due to the Sun's rotation relative to the stars
+    # A sidereal rotation is defined to be exactly 25.38 days
+    sidereal_lon = Longitude((obstime.jd - _time_first_rotation.jd) / 25.38 * 360*u.deg)
+
+    # Calculate the longitude of the Earth in de-tilted HCRS
+    lon_obstime = get_earth(obstime).hcrs.cartesian.transform(_sun_detilt_matrix) \
+                  .represent_as(SphericalRepresentation).lon.to('deg')
+
+    return Longitude(lon_obstime - _lon_first_rotation - sidereal_lon)
+
+
+def get_sun_P(time='now'):
+    """
+    Return the position (P) angle for the Sun at a specified time, which is the angle between
+    geocentric north and solar north as seen from Earth, measured eastward from geocentric north.
+    The range of P is +/-26.3 degrees.
+
+    Parameters
+    ----------
+    time : various
+        Time to use as `~astropy.time.Time` or in a parse_time-compatible format
+
+    Returns
+    -------
+    out : `~astropy.coordinates.Angle`
+        The position angle
+    """
+    obstime = _astropy_time(time)
+
+    geocentric = PrecessedGeocentric(equinox=obstime, obstime=obstime)
+
+    return _sun_north_angle_to_z(geocentric)
+
+
+def get_sunearth_distance(time='now'):
+    """
+    Return the distance between the Sun and the Earth at a specified time.
+
+    Parameters
+    ----------
+    time : various
+        Time to use as `~astropy.time.Time` or in a parse_time-compatible format
+
+    Returns
+    -------
+    out : `~astropy.coordinates.Distance`
+        The Sun-Earth distance
+    """
+    return get_earth(time).radius
+
+
+def get_sun_orientation(location, time='now'):
+    """
+    Return the orientation angle for the Sun from a specified Earth location and time.  The
+    orientation angle is the angle between local zenith and solar north, measured eastward from
+    local zenith.
+
+    Parameters
+    ----------
+    location : `~astropy.coordinates.EarthLocation`
+        Observer location on Earth
+    time : various
+        Time to use as `~astropy.time.Time` or in a parse_time-compatible format
+
+    Returns
+    -------
+    out : `~astropy.coordinates.Angle`
+        The orientation of the Sun
+    """
+    obstime = _astropy_time(time)
+
+    local_frame = AltAz(obstime=obstime, location=location)
+
+    return _sun_north_angle_to_z(local_frame)
+
+
+def _sun_north_angle_to_z(frame):
+    """
+    Return the angle between solar north and the Z axis of the provided frame's coordinate system
+    and observation time.
+    """
+    # Find the Sun center and Sun north in HGS at the frame's observation time
+    sun_center = SkyCoord(0*u.deg, 0*u.deg, 0*u.km, frame=HGS, obstime=frame.obstime)
+    sun_north = SkyCoord(0*u.deg, 90*u.deg, 690000*u.km, frame=HGS, obstime=frame.obstime)
```

Comment:
I am right in thinking that the actual value of radius doesn't matter here right? Maybe a comment to explain that?